src/components/tl/cuda/reduce_scatter/reduce_scatter_linear.c

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "reduce_scatterv/reduce_scatterv.h"
+#include "reduce_scatter/reduce_scatter.h"
+
+ucc_status_t ucc_tl_cuda_reduce_scatter_linear_init(ucc_base_coll_args_t *coll_args,
```

Comment:
Is it possible to call directly `ucc_tl_cuda_reduce_scatterv_linear_init`? I have the feeling that this function is nothing but a duplication of what is done in `ucc_tl_cuda_reduce_scatterv_linear_init` and `ucc_tl_cuda_reduce_scatterv_linear_start`

Reply:
ucc_tl_cuda_reduce_scatter_linear_init sets different pointers for get_count and get_offset. I can remove it and check collective type in ucc_tl_cuda_reduce_scatterv_linear init instead, but I wanted to explicitly show that reduce_scatter exists in TL CUDA. Otherwise, it will be hiddent inside reduce_scatterv

Reply:
Ok, I see. So do you think it would be a good idea that both `ucc_tl_cuda_reduce_scatter_linear_init` and `ucc_tl_cuda_reduce_scatterv_linear_init` call a common helper function that takes `get_count` and `get_offset` as arguments ?